src/hint_processor/builtin_hint_processor/vrf/pack_512.rs

```diff
@@ -19,13 +22,13 @@ def pack_512(d0, d1,d2,d3, num_bits_shift: int) -> int:
     return sum(limb << (num_bits_shift * i) for i, limb in enumerate(limbs))
 
 */
-#[allow(dead_code)]
-fn pack_512(limbs: &[Felt252; 4], num_bits_shift: usize) -> Felt252 {
-    let mut result = Felt252::zero();
-    for (i, limb) in limbs.iter().enumerate() {
-        result += limb << (num_bits_shift * i);
-    }
-    result
+fn pack_512(limbs: &[Felt252; 4], num_bits_shift: usize) -> BigInt {
+    #[allow(deprecated)]
+    limbs
+        .iter()
+        .enumerate()
+        .map(|(idx, value)| value.to_bigint().shl(idx * num_bits_shift))
```

Comment:
Unintuitively, `Felt252::to_bigint` alters the value and is probably not what you want here. Use `Felt252::to_biguint`. If you need it to be signed you can use `BigUint::to_bigint` on the result.
The reason for this is that it turns them into the signed version of the field, that is, it does `P - x` when `x > P / 2`.

Reply:
Thanks for this catch!